### PR TITLE
Prefix some part of MessageExtensions with _protobuf_.

### PR DESCRIPTION
**Pull request by thomasvl**
Working on marking things that aren't part of the public api
with _protobuf.

### CODE REVIEW

File: Sources/SwiftProtobuf/MessageExtension.swift [hunk 1]
```diff
@@ -19,22 +19,22 @@
 /// pieces.
 public protocol MessageExtensionBase {
     var fieldNumber: Int { get }
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/MessageExtension.swift]: I wasn't completely sure here, I'm guessing we want to expose the `fieldNumber` and `messageType`?

File: Sources/SwiftProtobuf/MessageExtension.swift [hunk 2]
```diff
@@ -19,22 +19,22 @@
...
 
 /// A "Message Extension" relates a particular extension field to
 /// a particular message.  The generic constraints allow
 /// compile-time compatibility checks.
 public class MessageExtension<FieldType: ExtensionField, MessageType: Message>: MessageExtensionBase {
     public let fieldNumber: Int
-    public var fieldNames: _NameMap.Names
+    public var _protobuf_fieldNames: _NameMap.Names
     public let messageType: Message.Type
     public let defaultValue: FieldType.ValueType
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/MessageExtension.swift]: Should `defaultValue` also get prefixed?

File: Sources/SwiftProtobuf/MessageExtension.swift [hunk 3]
```diff
@@ -19,22 +19,22 @@
...
     public init(fieldNumber: Int, fieldNames: _NameMap.Names, defaultValue: FieldType.ValueType) {
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/MessageExtension.swift]: I think this is one of the few initializers we have that really is just for the generated code, do we want to do anything with the naming of this?

File: Sources/SwiftProtobuf/MessageExtension.swift (hunk 1 above)
Comment (tbkka) [inline Sources/SwiftProtobuf/MessageExtension.swift]: That sounds right.

File: Sources/SwiftProtobuf/MessageExtension.swift (hunk 3 above)
Comment (allevato) [inline Sources/SwiftProtobuf/MessageExtension.swift]: The Swift stdlib prefixes the first argument label with `_` to indicate "public but private". You could prepend it with that, or with `_protobuf_` if you wanted to make it even uglier/scarier.

File: Sources/SwiftProtobuf/MessageExtension.swift (hunk 2 above)
Comment (tbkka) [inline Sources/SwiftProtobuf/MessageExtension.swift]: I don't think so:  `defaultValue` comes straight from the proto file so I think it's something we can commit to.

File: Sources/SwiftProtobuf/MessageExtension.swift (hunk 3 above)
Comment (tbkka) [inline Sources/SwiftProtobuf/MessageExtension.swift]: The convention I've seen elsewhere is to prefix the first argument label:
```
  public init(_protobuf_fieldNumber: Int, fieldNames: ...)
```

Comment (thomasvl) [inline Sources/SwiftProtobuf/MessageExtension.swift]: Do we do `_protobuf_` on the first arg (`fieldNumber`) or should we put it on `fieldNames` to match the properties in general?

Comment (thomasvl) [inline Sources/SwiftProtobuf/MessageExtension.swift]: New commit prefixes the first arg to init.

**Comment by thomasvl** [comment]
Added the prefix to the `set()` method also since it also seems like it should be internal.

**Pull request merged**